hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -578,6 +578,21 @@ public HoodieInstant transitionRestoreRequestedToInflight(HoodieInstant requeste
     return inflight;
   }
 
+  /**
+   * Transition Restore State from inflight to requested.
+   *
+   * @param inflightInstant inflight instant
+   * @return commit instant
+   */
+  public HoodieInstant transitionRestoreInflightToRequested(HoodieInstant inflightInstant) {
+    ValidationUtils.checkArgument(inflightInstant.getAction().equals(HoodieTimeline.RESTORE_ACTION), "Transition to requested attempted for a restore instant with diff action "
+        + inflightInstant);
+    ValidationUtils.checkArgument(inflightInstant.isInflight(), "Transition to requested attempted for an instant not in inflight state " + inflightInstant);
+    HoodieInstant requested = new HoodieInstant(State.REQUESTED, RESTORE_ACTION, inflightInstant.getTimestamp());
+    transitionState(inflightInstant, requested, Option.empty());
```

Comment:
not sure if we can do this. we can just delete the inflight and make progress. we don't want to requested to be overwritten just incase.